src/main/java/org/stellar/sdk/StrKey.java

```diff
@@ -49,6 +65,28 @@ public static String encodeStellarAccountId(AccountID accountID) {
         return String.valueOf(encoded);
     }
 
+    public static String encodeSignedPayload(SignedPayloadSigner signedPayloadSigner) {
+        if (signedPayloadSigner.getPayload().length > SIGNED_PAYLOAD_MAX_PAYLOAD_LENGTH) {
+            throw new FormatException("invalid payload length, must be less than " + SIGNED_PAYLOAD_MAX_PAYLOAD_LENGTH);
+        }
+        if (!signedPayloadSigner.getAccountId().getAccountID().getDiscriminant().equals(PublicKeyType.PUBLIC_KEY_TYPE_ED25519)) {
+            throw new FormatException("invalid payload signer, only ED25519 public key accounts are supported currently");
+        }
+        try {
+            SignerKey.SignerKeyEd25519SignedPayload xdrPayloadSigner = new SignerKey.SignerKeyEd25519SignedPayload();
+            xdrPayloadSigner.setPayload(signedPayloadSigner.getPayload());
+            xdrPayloadSigner.setEd25519(signedPayloadSigner.getAccountId().getAccountID().getEd25519());
+
+            ByteArrayOutputStream record = new ByteArrayOutputStream();
+            xdrPayloadSigner.encode(new XdrDataOutputStream(record));
+
+            char[] encoded = encodeCheck(VersionByte.SIGNED_PAYLOAD, record.toByteArray());
+            return String.valueOf(encoded);
+        } catch (Exception ex) {
+            throw new FormatException(ex.getMessage());
```

Comment:
Why rethrow when you can just let it bubble up?

Reply:
to muffle the checked exceptions which otherwise would require adding on a `throws` declaration on the method for each one, which then forces the callers to specifically handle it, it's mostly for caller convenience.

Reply:
makes sense, java-isms.. :+1: